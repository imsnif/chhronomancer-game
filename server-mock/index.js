import playerStore from '../stores/player'
import timelineStore from '../stores/timeline'
import powerStore from '../stores/power'
import statsStore from '../stores/stats'

const timelines = [
  {name: 'Timeline 1', type: 'steal', players: []},
  {name: 'Timeline 2', type: 'assist', players: []},
  {name: 'Timeline 3', type: 'prevent', players: []},
  {name: 'Timeline 4', type: 'reset', players: []}
]

const players = [
  {
    id: 1,
    name: 'Aram',
    items: [
      {name: 'assist', source: false},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'reset', source: 'Timeline 4'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: false},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_me_s.png')
  },
  {
    id: 2,
    name: 'Tom',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 1'},
      {name: 'unlock', source: false}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_tom_s.png')
  },
  {
    id: 3,
    name: 'Gondollieri',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: 'Timeline 3'},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 3'},
      {name: 'unlock', source: 'Timeline 1'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_gondollieri_s.png')
  },
  {
    id: 4,
    name: 'Simmons',
    items: [
      {name: 'assist', source: 'Timeline 2'},
      {name: 'prevent', source: false},
      {name: 'reset', source: 'Timeline 3'},
      {name: 'steal', source: 'Timeline 1'},
      {name: 'lock', source: 'Timeline 2'},
      {name: 'unlock', source: 'Timeline 2'}
    ],
    actions: 10,
    image: require('../assets/placeholders/green_simmons_s.png')
  }
]

const powers = [
  {
    playerId: 1,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 1,
    timelineName: 'Timeline 2',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 400000,
    target: {
      type: 'timeline',
      name: 'Timeline 2'
    },
    allies: [
      {id: 1, score: 8},
      {id: 2, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 9}
    ]
  },
  {
    playerId: 2,
    timelineName: 'Timeline 1',
    name: 'Resetting',
    startTime: Date.now(),
    endTime: Date.now() + 1000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 2, score: 1},
      {id: 1, score: 5}
    ],
    enemies: [
      {id: 3, score: 1},
      {id: 4, score: 2}
    ]
  },
  {
    playerId: 3,
    timelineName: 'Timeline 1',
    name: 'Combining',
    startTime: Date.now(),
    endTime: Date.now() + 2000000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 3, score: 1},
      {id: 1, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 4, score: 5}
    ]
  },
  {
    playerId: 4,
    timelineName: 'Timeline 1',
    name: 'Locking',
    startTime: Date.now(),
    endTime: Date.now() + 100000,
    target: {
      type: 'timeline',
      name: 'Timeline 1'
    },
    allies: [
      {id: 4, score: 7},
      {id: 2, score: 2}
    ],
    enemies: [
      {id: 2, score: 2},
      {id: 1, score: 5}
    ]
  }
]

export default function mockServer () {
  timelines.forEach((t) => timelineStore.addTimeline(t))
  players.forEach(p => {
    playerStore.addPlayer(p)
  })
  statsStore.changePlayerId(1)
  timelineStore.addPlayer('Timeline 1', 1)
  timelineStore.addPlayer('Timeline 1', 2)
  timelineStore.addPlayer('Timeline 2', 2)
  timelineStore.addPlayer('Timeline 3', 2)
  timelineStore.addPlayer('Timeline 1', 3)
  timelineStore.addPlayer('Timeline 2', 3)
  timelineStore.addPlayer('Timeline 3', 3)
  timelineStore.addPlayer('Timeline 1', 4)
  timelineStore.addPlayer('Timeline 3', 4)
  powers.forEach(p => powerStore.addPower(p))
  timelineStore.changeTimelineLock('Timeline 1', true)
}
