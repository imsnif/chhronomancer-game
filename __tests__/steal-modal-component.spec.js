import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

beforeEach(() => {
  jest.useFakeTimers()
})

afterEach(() => {
  const timelineStore = require('../stores/timeline').default
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
  timelineStore.clearAllModals()
})

test('Steal Modal component => renders correctly when open', () => {
  const timelineName = 'Timeline 1'
  const StealModal = require('../components/steal-modal').default
  require('../server-mock') // TODO: export a function like a normal Human being
  const timelineStore = require('../stores/timeline').default
  timelineStore.addModal(timelineName, {
    modalType: 'steal',
    itemName: 'assist',
    playerName: 'foo'
  })
  const tree = renderer.create(
    <StealModal timelineName={timelineName} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Steal Modal component => renders correctly when closed', () => {
  const timelineName = 'Timeline 1'
  const StealModal = require('../components/steal-modal').default
  require('../server-mock') // TODO: export a function like a normal Human being
  const tree = renderer.create(
    <StealModal timelineName={timelineName} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
