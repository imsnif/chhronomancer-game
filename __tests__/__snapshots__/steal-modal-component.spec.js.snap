// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steal Modal component => renders correctly when closed 1`] = `<View />`;

exports[`Steal Modal component => renders correctly when open 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "zIndex": 2,
      },
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "black",
              "opacity": 0.5,
            },
          ]
        }
      />
    </View>
    <View
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#141313",
          "height": "50%",
          "transform": Array [
            Object {
              "translateY": 1334,
            },
            Object {
              "translateX": 0,
            },
          ],
          "width": "90%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": "10%",
            "resizeMode": "stretch",
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": "80%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#329932",
                  "flex": 0,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            Would you like to steal
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 16,
                  "marginRight": 0,
                  "width": 16,
                }
              }
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#329932",
                  "flex": 0,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            from
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#329932",
                  "flex": 0,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 45.911331934501746,
                  "textAlign": "center",
                  "width": "100%",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            foo
            ?
          </Text>
          <View
            style={
              Object {
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#141313",
                        "borderColor": "#329932",
                        "borderRadius": 0,
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "center",
                        "width": "100%",
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#329932",
                        "flex": 1,
                        "fontFamily": "telegrama_raw",
                        "textAlign": "center",
                      }
                    }
                  >
                    Yes
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginLeft": 2,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#141313",
                        "borderColor": "#329932",
                        "borderRadius": 0,
                        "borderWidth": 1,
                        "flex": 1,
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "center",
                        "width": "100%",
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#329932",
                        "flex": 1,
                        "fontFamily": "telegrama_raw",
                        "textAlign": "center",
                      }
                    }
                  >
                    No
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Image
        source={1}
        style={
          Object {
            "height": "10%",
            "resizeMode": "stretch",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</View>
`;
